File: src/FoodStuffs.Web/ClientApp/src/models/api/recipes/GetRecipeResponse.js
export default class GetRecipeResponse {
  constructor() {
    this.id = 0;
    this.name = '';
    this.ingredients = '';
    this.directions = '';
    this.cookTimeMinutes = null;
    this.prepTimeMinutes = null;
    this.pinnedImageId = null;
    this.categories = [];
    this.images = [];
    this.createdOn = new Date();
    this.createdBy = '';
    this.modifiedOn = new Date();
    this.modifiedBy = '';
  }
}

File: src/FoodStuffs.Model/Events/Recipes/GetRecipeResponse.cs
namespace FoodStuffs.Model.Events.Recipes;

public record GetRecipeResponse(
    int Id,
    string Name,
    string Directions,
    int? PrepTimeMinutes,
    int? CookTimeMinutes,
    bool IsForMealPlanning,
    string CreatedBy,
    DateTimeOffset CreatedOn,
    string ModifiedBy,
    DateTimeOffset ModifiedOn,
    string Slug,
    string? DefaultImage,
    string? PinnedImage,
    List<string> Images,
    List<string> Categories,
    List<GetRecipeResponseIngredient> Ingredients,
    List<GetRecipeResponseShoppingItem> ShoppingItems);
